reconcile/aus/base.py

```diff
@@ -771,6 +856,16 @@ def calculate_diff(
     now = datetime.utcnow()
     gates = get_version_gates(ocm_api)
     for spec in desired_state.specs:
+        # Upgrading node pools, only required for Hypershift clusters
```

Comment:
on the previous block, shouldn't we set mutexes when there are ongoing nodepool upgrades as well?

Reply:
Very good point. I think I have to change quite some bits in the current PR to meet this.